Sources/SwiftBSON/BSONDocument+Sequence.swift

```diff
@@ -29,7 +29,11 @@ extension BSONDocument: Sequence {
     public func map<T>(
         _ transform: (Element) throws -> T
     ) rethrows -> [T] {
-        try AnySequence(self).map(transform)
```

Comment:
This was calling `count` a lot, which now requires a full walk over the document. To avoid any issues, I just did a naive manual implementation of `map`.